packages/transaction-controller/src/utils/batch.ts

```diff
@@ -410,14 +412,25 @@ async function addTransactionBatchWithHook(
   const collectHook = new CollectPublishHook(transactionCount);
   try {
     if (requireApproval && useHook) {
+      const { gasLimit, transactions: transactionsWithGas } =
```

Comment:
Minor, as this logic expands, does it warrant a `updateGasProperties` function or maybe a `prepareApprovalData`?

Reply:
Created the `prepareApprovalData` to move the logic and expand it into a separate function.